src/unix/fs.c

```diff
@@ -1076,18 +1077,78 @@ static ssize_t uv__fs_write(uv_fs_t* req) {
 }
 
 static ssize_t uv__fs_copyfile(uv_fs_t* req) {
+  uv_file dstfd;
+  int dst_flags;
+#if defined(__APPLE__) && !TARGET_OS_IPHONE
+  /* On macOS, use the native copyfile(3). */
+  static int can_clone;
+  copyfile_flags_t flags;
+  char buf[64];
+  size_t len;
+  int major;
+
+  flags = COPYFILE_ALL;
+  dst_flags = O_WRONLY | O_CREAT | O_TRUNC;
+
+  /* Copyfile has its own, but let's emulate everything */
+  if (req->flags & UV_FS_COPYFILE_EXCL)
+    dst_flags |= O_EXCL;
+
+  /* Check OS version. Cloning is only supported on macOS >= 10.12. */
+  if (req->flags & UV_FS_COPYFILE_FICLONE_FORCE) {
+    if (can_clone == 0) {
+      len = sizeof(buf);
+      if (sysctlbyname("kern.osrelease", buf, &len, NULL, 0))
+        return UV__ERR(errno);
+
+      if (1 != sscanf(buf, "%d", &major))
+        abort();
+
+      can_clone = -1 + 2 * (major >= 16);  /* macOS >= 10.12 */
+    }
+
+    if (can_clone < 0)
+      return UV_ENOSYS;
+  }
+
+  /* copyfile() simply ignores COPYFILE_CLONE if it's not supported. */
+  if (req->flags & UV_FS_COPYFILE_FICLONE)
+    flags |= 1 << 24;  /* COPYFILE_CLONE */
+
+  if (req->flags & UV_FS_COPYFILE_FICLONE_FORCE)
+    flags |= 1 << 25;  /* COPYFILE_CLONE_FORCE */
+
+  /* Respect target permissions by trying to open. */
+  dstfd = uv_fs_open(NULL,
+                     &fs_req,
+                     req->new_path,
+                     dst_flags,
+                     src_statsbuf.st_mode,
+                     NULL);
+  uv_fs_req_cleanup(&fs_req);
+
+  if (dstfd < 0) {
+    return -dstfd;
```

Comment:
```suggestion
    return dstfd;
```
(it's already a libuv error code)